src/status_im/ui/screens/syncing/sheets/enter_password/views.cljs

```diff
@@ -0,0 +1,88 @@
+(ns status-im.ui.screens.syncing.sheets.enter-password.views
+  (:require [quo.react-native :as rn]
+            [status-im.ui.screens.syncing.sheets.enter-password.styles :as styles]
+            [re-frame.core :as re-frame]
+            [quo.core :as quo]
+            [status-im.i18n.i18n :as i18n]
+            [status-im.ui.components.react :as react]
+            [status-im.ui.components.toolbar :as toolbar]
+            [clojure.string :as string]
+            [taoensso.timbre :as log]
+            [status-im.utils.security :as security]
+            [status-im.ethereum.core :as ethereum]
+            [status-im.utils.fx :as fx]
+            [status-im.native-module.core :as status]
+            [quo2.components.info.information-box :as information-box]
+            [status-im.ui.components.qr-code-viewer.views :as qr-code-viewer]
+            [status-im.utils.handlers :refer [>evt]]))
+
+(def entered-password (atom ""))
+(def connection-string  (atom ""))
+
+(defn qr-code-view-with-connection-string [connection-string]
+  (log/info "connection-string ===> " connection-string)
+  (let [window-width       @(re-frame/subscribe [:dimensions/window-width])
+        valid-cs?          (string/starts-with? connection-string "cs")]
+    [:<>
+     (if valid-cs?
+       [rn/view {:style styles/body-container}
+        [rn/text {:style styles/header-text} "Sync code generated"]
+        [qr-code-viewer/qr-code-view (* window-width 0.808) connection-string]
```

Comment:
why are we using this magic number here? Is there not another approach we can take to get the view width, perhaps using `on-layout`?

Reply:
hmm this was made 3 months ago, the idea was to swiftly get the UI up and running, I shall refactor :)
Will look into `on-layout` , thanks for the suggestion

Reply:
cool, that's what I did in this p.r: https://github.com/status-im/status-mobile/pull/14444/files#diff-427a453633106ea987e59e011963a453be85a66a9cb88f34359346c35287e382R82

Reply:
@J-Son89  : `[qr-code-viewer/qr-code-view` expects a number to be passed for drawing up the QR code, I wished it accepted percentage values but yeah it needs a number to draw up the QR SVG.
So the calculation i do Is basically I get the 80% window width from Dimensions( React Native ).

Is there any benefit of using on-layout over this approach? if not I would like to keep this as it is,
Also note that this view will undergo some changes as and when the functionality reaches completion.

Also one more thing to consider is that once QR codes are implemented in status-go side this component would again become redundant :)